### PR TITLE
[fix](inverted index) fix the issue where need_read_data does not handle non-pushdown conditions

### PR DESCRIPTION
**Pull request by zzzxl1993**
…dle non-pushdown conditions

## Proposed changes

1. Fixed the previous optimization of https://github.com/apache/doris/pull/35346 not handling non-pushdown conditions.

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run cloud_p1

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -343,6 +343,34 @@ Status SegmentIterator::_init_impl(const StorageReadOptions& opts) {
...
     }
+
+    // find columns that definitely require reading data, such as functions that are not pushed down.
+    {
+        std::set<std::string> push_down_preds;
+        for (auto* pred : _col_predicates) {
+            if (!_check_apply_by_inverted_index(pred)) {
+                continue;
+            }
+            push_down_preds.insert(_gen_predicate_result_sign(pred));
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:354]: Pred may not be evaluated by index even if _check_apply_by_inverted_index(pred) return true due to skip index based on hit ratio. So push_down_preds.contains(column_sign) can not guarantee that the pred for column_sing can be skipped.

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -984,7 +1012,7 @@ std::string SegmentIterator::_gen_predicate_result_sign(ColumnPredicate* predica
 
     std::string col_name = column_desc->name();
 
-    if (column_desc->path() != nullptr) {
+    if (column_desc->path() != nullptr && !_storage_name_and_type.empty()) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1015]: Is this another problem?

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:354]: This section only calculates which conditions definitely require data reading, the conditions that do not require data reading are computed in other logical calculations.

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**